crates/accelerate/src/commutation_checker.rs

```diff
@@ -0,0 +1,725 @@
+        let qargs2 =
+            qargs2.map_or_else(|| Ok(PyTuple::empty_bound(py)), PySequenceMethods::to_tuple)?;
+        let cargs2 =
+            cargs2.map_or_else(|| Ok(PyTuple::empty_bound(py)), PySequenceMethods::to_tuple)?;
+        self.commute_nodes(
+            py,
+            &DAGOpNode {
+                instruction: CircuitInstruction::py_new(
+                    op1,
+                    Some(qargs1.into_any()),
+                    Some(cargs1.into_any()),
+                )?,
+                sort_key: PyString::new_bound(py, "do_not_use").into_any().into(),
+            },
+            &DAGOpNode {
+                instruction: CircuitInstruction::py_new(
+                    op2,
+                    Some(qargs2.into_any()),
+                    Some(cargs2.into_any()),
+                )?,
+                sort_key: PyString::new_bound(py, "do_not_use").into_any().into(),
+            },
+            max_num_qubits,
+        )
+    }
+
+    #[pyo3(signature=())]
+    fn num_cached_entries(&self) -> usize {
+        self.current_cache_entries
+    }
+    #[pyo3(signature=())]
+    fn clear_cached_commutations(&mut self) {
+        self.clear_cache()
+    }
+
+    fn __getstate__(&self, py: Python) -> PyResult<Py<PyDict>> {
+        let out_dict = PyDict::new_bound(py);
+        out_dict.set_item("cache_max_entries", self.cache_max_entries)?;
+        out_dict.set_item("current_cache_entries", self.current_cache_entries)?;
+        out_dict.set_item("_cache_miss", self._cache_miss)?;
+        out_dict.set_item("_cache_hit", self._cache_hit)?;
+        out_dict.set_item("cache", self.cache.clone())?;
+        out_dict.set_item("library", self.library.library.to_object(py))?;
+        out_dict.set_item("gates", self.gates.clone())?;
+        Ok(out_dict.unbind())
+    }
+
+    fn __setstate__(&mut self, py: Python, state: PyObject) -> PyResult<()> {
+        let dict_state = state.downcast_bound::<PyDict>(py)?;
+        self.cache_max_entries = dict_state
+            .get_item("cache_max_entries")?
+            .unwrap()
+            .extract()?;
+        self.current_cache_entries = dict_state
+            .get_item("current_cache_entries")?
+            .unwrap()
+            .extract()?;
+        self._cache_miss = dict_state.get_item("_cache_miss")?.unwrap().extract()?;
+        self._cache_hit = dict_state.get_item("_cache_hit")?.unwrap().extract()?;
+        self.library = CommutationLibrary {
+            library: dict_state.get_item("library")?.unwrap().extract()?,
+        };
+        self.cache = dict_state.get_item("cache")?.unwrap().extract()?;
+        self.gates = dict_state.get_item("gates")?.unwrap().extract()?;
+        Ok(())
+    }
+}
+
+impl CommutationChecker {
+    #[allow(clippy::too_many_arguments)]
+    fn commute_inner(
+        &mut self,
+        py: Python,
+        instr1: &CircuitInstruction,
+        qargs1: &[u32],
+        cargs1: &[u32],
```

Comment:
As I mentioned above I think using

```suggestion
        qargs1: &[Qubit],
        cargs1: &[Clbit],
```

here is a better choice.

Reply:
done in 454208cd9